modules/library/main/src/main/java/org/geotools/geometry/jts/JTS.java

```diff
@@ -1685,23 +1688,61 @@ public static CoordinateReferenceSystem getCRS(Geometry geometry) {
             if (crs != null) {
                 return crs;
             }
+        } else if (geometry.getSRID() > 0) {
+            try {
+                return CRS.decode("EPSG:" + geometry.getSRID());
+            } catch (Exception e) {
+                return null;
+            }
         }
         return null;
     }
 
     /**
-     * Set the CRS of a geometry.
+     * Set the CRS of a geometry is {@link Geometry#getUserData()}.
+     *
+     * <p>Stored in {@link Geometry#getUserData()} as a {@code CoordinateReferenceSystem} directly,
+     * or as a {@code Map<Class,Value>}.
+     *
+     * <p>Stored in {@link Geometry#getSRID()} as EPSG code.
      *
      * @param geometry the geometry
      * @param crs the CRS
      */
     public static void setCRS(Geometry geometry, CoordinateReferenceSystem crs) {
-        Object obj = geometry.getUserData();
-        Map<Object, Object> userData = new HashMap<>();
-        if (obj != null && obj instanceof Map) {
-            userData.putAll((Map<?, ?>) obj);
+        if (crs != null) {
+            try {
+                Integer srid = CRS.lookupEpsgCode(crs, false);
+                if (srid != null) {
+                    geometry.setSRID(srid);
+                }
+            } catch (FactoryException noSRID) {
+                geometry.setSRID(0);
+            }
```

Comment:
Oh gosh no no no... the lookup is not free, doing it for every geometry is criminal performance-wise, but more importantly, usage of SRID should be banned from all the codebase, with the exception of places where it's the only option (spatial databases), because it does not carry an authority (since the IAU work we live in a post-EPSG world).

Reply:
I was gathering up this code from  CRSEvaluator that did this, in part to catch assumptions like this :)